### DISCUSSION TITLE
6 month sell through rates and ratios

### DISCUSSION THREAD

**Discussion by Bricktastics123** [rgriebl/brickstore#751] [Ideas]
Is it possible to add a column to the working files that enables the display of the sell through rate on parts. Ie: Current Inventory/Sold inventory and display as a ratio. If this could be displayed it would really help my team understand if a part is likely to sell if listed and over what time frame and we could quickly make decisions based on this factor and how it would relate to allocating shelf space to parts that are very unpopular.

Thankyou!

**Comment by BEdders1987** [comment]
This would be a greatly desired feature. Being able to see the sell through rate would allow user to determine if a set is going to be profitable to part out.

**Comment by IllHyperion** [comment]
I agree! It would be very helpful.

I've been working on an application that finds profit and sell through rates on Lego.com pick a brick items but it could be adapted to use any item on brick link

The private affiliate api that BrickStore uses for its pricing date does give the information needed to calculate the sell through.

I've just made a simple script for myself which can, given a .bsx file, update the comments column for each item with the sell through rate. So it's definitely possible.

**Reply by 62bricks** [reply]
I assume the affiliate API is different than the store API? Where do you get access?

**Reply by IllHyperion** [reply]
The affiliate api isn't publicly available, I just noticed how it works from some of the code from BrickStore. The main difference is that you can request multiple items at once, instead of just 1 at a time with the public api. And also there is no 5000 request a day cap.

**Reply by 62bricks** [reply]
Exactly what I need. I've tried getting the price guide data from the Brickstore cache, but I can't figure out how to decode the blob (I think it's a blob) in the sqlite file.

**Reply by IllHyperion** [reply]
I added you to a private repo of one of the projects I'm working on the uses the affiliate api, I guess I'd just ask you to not share the api key around since it's hard coded in right now.

So you can take a look around (mostly in src/database/BLPricesV2.py) for the stuff with the api.

**Reply by 62bricks** [reply]
Thanks! I'll take a look.

**Reply by m2gChristian** [reply]
> I added you to a private repo of one of the projects I'm working on the uses the affiliate api, I guess I'd just ask you to not share the api key around since it's hard coded in right now.
> 
> So you can take a look around (mostly in src/database/BLPricesV2.py) for the stuff with the api.

This sounds interresting. I also started with (new to) python to do some basic jobs to automate with bsx files. I also wanted to implement something to automate price updates in future. What I also find interesting is to work with BrickStore cache, but I did not get far on my first research. What I think it might also be interesting is the BrickStore extensions(but I am not sure if I want to learn something new next to python with my limited time for this hobby). I did not find projects depending extensions yet. Did you also look at this? It should have database (full catalog from cache) and price guide objects.

Regarding on the API Key I started to save it in a local config.ini and than you can exclude from commit. As I only commit local I don't have issues there but probably I have it easy in future cases..

**Reply by steve-parkinson-bl** [reply]
I've managed to work out sell thru rates but can the affiliate API give regional prices?
